tests/functional/dispersion/dispersion.py

```diff
@@ -0,0 +1,236 @@
+        protons={"charge": 1, "density": density, **vvv}
+    )
+
+    ElectronModel(closure="isothermal", Te=0.12)
+
+
+    sim = ph.global_vars.sim
+
+    timestamps = np.arange(0, sim.final_time +sim.time_step, sim.time_step)
+
+    for quantity in ["E", "B"]:
+        ElectromagDiagnostics(
+            quantity=quantity,
+            write_timestamps=timestamps,
+            compute_timestamps=timestamps,
+        )
+
+
+
+
+def prescribedModes(): #TODOit
+
+    # in this configuration there user prescribed
+    # wavelength, at which more energy is thus expected
+    # than in modes naturally present in the simulation noise
+
+    Simulation(
+        smallest_patch_size=20,
+        largest_patch_size=20,
+
+        # the following time step number
+        # and final time mean that the
+        # smallest frequency will be 2/100
+        # and the largest 2/dt  = 2e3
+        time_step_nbr=100000,
+        final_time=100.,
+
+        boundary_types="periodic",
+
+        # smallest wavelength will be 2*0.2=0.4
+        # and largest 50
+        cells=500,
+        dl=0.2,
+        diag_options={"format": "phareh5", "options": {"dir": "dispersion",
+                                                       "mode":"overwrite"}}
+    )
+
+    def density(x):
+        return 1.
+
+
+    def by(x):
+        from pyphare.pharein.global_vars import sim
+        L = sim.simulation_domain()
+        return 0.1*np.cos(2*np.pi*x/L[0])
+
+
+    def bz(x):
+        from pyphare.pharein.global_vars import sim
+        L = sim.simulation_domain()
+        return 0.1*np.sin(2*np.pi*x/L[0])
+
+
+    def bx(x):
+        return 1.
+
+
+    def vx(x):
+        return 0.
+
+
+    def vy(x):
+        from pyphare.pharein.global_vars import sim
+        L = sim.simulation_domain()
+        return 0.1*np.cos(2*np.pi*x/L[0])
+
+    def vz(x):
+        from pyphare.pharein.global_vars import sim
+        L = sim.simulation_domain()
+        return 0.1*np.sin(2*np.pi*x/L[0])
+
+
+    def vthx(x):
+        return 0.01
+
+
+    def vthy(x):
+        return 0.01
+
+
+    def vthz(x):
+        return 0.01
+
+
+    vvv = {
+        "vbulkx": vx, "vbulky": vy, "vbulkz": vz,
+        "vthx": vthx, "vthy": vthy, "vthz": vthz
+    }
+
+    MaxwellianFluidModel(
+        bx=bx, by=by, bz=bz,
+        protons={"charge": 1, "density": density, **vvv, "init": {"seed": 1337}}
+    )
+
+    ElectronModel(closure="isothermal", Te=0.12)
+
+
+
+    sim = ph.global_vars.sim
+
+    timestamps = np.arange(0, sim.final_time +sim.time_step, sim.time_step)
+
+
+
+    for quantity in ["E", "B"]:
+        ElectromagDiagnostics(
+            quantity=quantity,
+            write_timestamps=timestamps,
+            compute_timestamps=timestamps,
+        )
+
+
+    for quantity in ["density", "bulkVelocity"]:
+        FluidDiagnostics(
+            quantity=quantity,
+            write_timestamps=timestamps,
+            compute_timestamps=timestamps,
+            )
+
+
+
```

Comment:
ok so the first test is for all modes with k along b0, while the second one is with velocity and magnetic field in hase... so I would say only alfven no ? might need to be written in comments ?